src/controllers/kafka-demo.controller.ts

```diff
@@ -239,4 +240,44 @@ export class KafkaDemoController {
       });
     });
   }
+
+  /**
+   * Consume messages of a given topic
+   * @param topic The topic name
+   */
+  @get('/topics/{topic}/messages', {
+    responses: {
+      '200': {
+        'text/event-stream': {
+          schema: {
+            type: 'string',
+          },
+        },
+      },
+    },
+  })
+  async consumeMessagesOnTopics(
+    @param.path.string('topic') topic: string,
+    @param.query.string('limit') limit: number,
+    @inject(RestBindings.Http.RESPONSE) response: Response,
+  ) {
+    let client = new KafkaClient();
+    let consumer = new Consumer(client, [{topic: topic}], {autoCommit: false});
+
+    limit = +limit || 5;
+    response.setHeader('Cache-Control', 'no-cache');
+    response.contentType('text/event-stream');
+
+    let count = 0;
+    consumer.on('message', message => {
+      count++;
+      response.write(`id: ${message.offset}\n`);
+      response.write('event: message\n');
+      response.write(`data: ${JSON.stringify(message)}\n`);
+      if (count >= limit) {
+        response.end();
+      }
+    });
+    return response;
```

Comment:
The logic to fetch a number of messages for a consumer is duplicate. Can we refactor it into a private method?

Reply:
added `writeMessageToResponse` function.